src/rewriters/ArithmeticEqualityRewriter.h

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Martin Blicha on 22.02.21.
+//
+
+#ifndef OPENSMT_ARITHMETICEQUALITYREWRITER_H
+#define OPENSMT_ARITHMETICEQUALITYREWRITER_H
+
+#include "Rewriter.h"
+
+class EqualityRewriterConfig : DefaultRewriterConfig {
+    LALogic & logic;
+
+public:
+    EqualityRewriterConfig(LALogic & logic): logic(logic) {}
+
+    bool previsit(PTRef term) override { return logic.hasSortBool(term); }
+
+    PTRef rewrite(PTRef term) override {
+        if (logic.isNumEq(term)) {
+            Pterm const  & p = logic.getPterm(term);
+            PTRef a1 = p[0];
+            PTRef a2 = p[1];
+            vec<PTRef> args;
+            args.push(a1); args.push(a2);
+            PTRef i1 = logic.mkNumLeq(args);
+            PTRef i2 = logic.mkNumGeq(args);
+            logic.markSplitInequality(i1); // MB: this information should not be stored in Logic!
```

Comment:
This information is only used in Lookahead solving. I think we should find a way to move it out of Logic.

Reply:
Ok, I'll commit a PR to fix this once we merge the current one.